dm/worker/log_test.go

```diff
@@ -0,0 +1,308 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"testing"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/dm/dm/pb"
+)
+
+func TestLog(t *testing.T) {
+	TestingT(t)
+}
+
+type testLog struct{}
+
+var _ = Suite(&testLog{})
+
+func (t *testLog) TestPointer(c *C) {
+	p := &Pointer{
+		Location: 2,
+	}
+
+	np := new(Pointer)
+
+	bs, _ := p.MarshalBinary()
+	c.Assert(np.UnmarshalBinary(bs), IsNil)
+	c.Assert(np, DeepEquals, p)
+	c.Assert(np.UnmarshalBinary([]byte("xx")), NotNil)
+}
+
+func (t *testLog) TestLoadHandledPointer(c *C) {
+	p, err := LoadHandledPointer(nil)
+	c.Assert(err, Equals, ErrInValidHandler)
+	c.Assert(p.Location, Equals, int64(0))
+
+	db, _ := testSetUpDB(c)
+	defer db.Close()
+	p, err = LoadHandledPointer(db)
+	c.Assert(err, IsNil)
+	c.Assert(p.Location, Equals, int64(0))
+
+	p = Pointer{
+		Location: 1,
+	}
+
+	bs, _ := p.MarshalBinary()
+	c.Assert(db.Put(HandledPointerKey, bs, nil), IsNil)
+	p, err = LoadHandledPointer(db)
+	c.Assert(err, IsNil)
+	c.Assert(p.Location, Equals, int64(1))
+
+	c.Assert(db.Put(HandledPointerKey, []byte("xx"), nil), IsNil)
+	_, err = LoadHandledPointer(db)
+	c.Assert(err, ErrorMatches, ".*not valid length data as.*")
+}
+
+func (t *testLog) TestTaskLogKey(c *C) {
+	var id int64 = 1
+	idc, err := DecodeTaskLogKey(EncodeTaskLogKey(id))
+	c.Assert(err, IsNil)
+	c.Assert(idc, DeepEquals, id)
+
+	_, err = DecodeTaskLogKey([]byte("xx"))
+	c.Assert(err, ErrorMatches, ".*not valid length data as.*")
+}
+
+func (t *testLog) TestTaskLog(c *C) {
+	logger := new(Logger)
```

Comment:
how about providing a function like `NewLogger` to create a `Logger` object, so if more fields added to `Logger`, less code needed to change.

Reply:
now it's enough, we can do it while we need it